service/history/workflow/task_generator.go

```diff
@@ -470,7 +470,7 @@ func (r *TaskGeneratorImpl) GenerateActivityRetryTasks(
 	r.mutableState.AddTasks(&tasks.ActivityRetryTimerTask{
 		// TaskID is set by shard
 		WorkflowKey:         r.mutableState.GetWorkflowKey(),
-		Version:             activity.Version,
+		Version:             r.mutableState.GetCurrentVersion(),
 		VisibilityTimestamp: activity.ScheduledTime.AsTime(),
```

Comment:
This field needs to be fixed as well. I think right now it's using the ScheduledTime of the previous attempt (the one in activityInfo before it's updated).